### ISSUE TITLE
Enhance the accessibility of `MonthPicker` (`@yamada-ui/calendar`) component

### ISSUE THREAD

**Issue by koralle** [yamada-ui/yamada-ui#1733]
### 1. Accessibility roles

* [x] Please refer to the image.

![month-picker](https://github.com/yamada-ui/yamada-ui/assets/33865215/c49c7ca4-421f-42c4-a02f-f11007ac7cca)

### 2. `aria-controls`

* [x] An element with the combobox role must have in its aria-controls attribute the ID of the corresponding popup element (the element with the dialog role).

### 3. `aria-haspopup`

* [x] You must set `aria-haspopup="dialog"` on elements with the `combobox` role.

### 4. `aria-expanded`

* [x] When an element with the `dialog` role is hidden, the element with the `combobox` role should have the `aria-expanded` attribute set to `false`.
* [x] When an element with the `dialog` role is visible, the element with the `combobox` role should have the `aria-expanded` attribute set to `true`.

~~### 5. `aria-activedescendant`~~

* [ ] ~~While a descendant element of the element with the `dialog` role is focused, the element with the `combobox` role must have the ID of the focused element as the value of its `aria-activedescendant` attribute.~~

### 6. `aria-label` / `aria-labelledby` (on the element with `combobox` role)

* [x] Please refer to the following flowchart.

```
Does the combobox have a visible label?
    ├──> No: `combobox` element has a label provided by `aria-label` . 
    └──> Yes
           └──> Is the `combobox` element an HTML element that can be labelled using the HTML `label` element (e.g., the input element)?
                  ├──> No: `combobox` element has `aria-labelledby` set to a value that refers to the labelling element. 
                  └──> Yes: `combobox` is labeled using the `label` element. 
```

### 7. `aria-modals`

* [x] You must set aria-modal="true" on elements with the dialog role.

### 8. `aria-live`

* [x] Please refer to the image.

![month-picker_1](https://github.com/yamada-ui/yamada-ui/assets/33865215/8343f444-b948-42fe-a149-6da1f6e74439)

### 9. `aria-selected`

* [x] You must add `aria-selected="true"` to the cell of the currently selected date (which is implemented as a `button` element in Yamada UI).

#### When the focus is within the popup...

* [x] If possible, it would be better to clear the current value instead of accepting it when the `Space` key is pressed.

## References

* [Date Picker Combobox Example | APG | WAI | W3C](https://www.w3.org/WAI/ARIA/apg/patterns/combobox/examples/combobox-datepicker/)
* [MonthPicker - Yamada UI](https://yamada-ui.com/components/forms/month-picker)
* [ARIA ライブリージョン - アクセシビリティ | MDN](https://developer.mozilla.org/ja/docs/Web/Accessibility/ARIA/ARIA_Live_Regions)

### Problem Statement/Justification

We should prioritize web accessibility to ensure our UI components are usable by everyone, including people with disabilities. This inclusivity enhances user experience and complies with legal standards, broadening Yamada UI's user base.

### Are you willing to participate in implementing this feature and create a pull request include the implementation?

None

**Comment by cidkumagai** [comment]
I'll try it.

**Comment by koralle** [comment]
Sorry, the `aria-activedescendant` attribute is not needed when the popup is a `dialog`.

> When a descendant of a listbox, grid, or tree popup is focused, DOM focus remains on the combobox and the combobox has [aria-activedescendant](https://w3c.github.io/aria/#aria-activedescendant) set to a value that refers to the focused element within the popup.
